homeassistant/components/onewire/__init__.py

```diff
@@ -1 +1,37 @@
 """The 1-Wire component."""
+import asyncio
+
+import voluptuous as vol
+
+from .const import DOMAIN, SUPPORTED_PLATFORMS
+
+CONFIG_SCHEMA = vol.Schema(
+    {DOMAIN: vol.Schema({}, extra=vol.ALLOW_EXTRA)}, extra=vol.ALLOW_EXTRA
+)
+
+
+async def async_setup(hass, config):
+    """Old way of setting up 1-Wire integrations."""
```

Comment:
The `async_setup` function is still needed. So it's not really true that it's the old way.